script/lib/filesystem.py

```diff
@@ -51,15 +51,17 @@ def download_and_extract(destination, url):
     with contextlib.closing(urllib2.urlopen(url)) as u:
       total = int(u.headers['content-length'])
       done = 0
+      last_length = 0
       while True:
         chunk = u.read(1024)
         done += len(chunk)
         if not len(chunk):
           break
-        sys.stderr.write('                                   \r')
-        percent = str(round(float(done) / float(total), 4) * 100) + '%'
+        percent = '{0:.2f}%%'.format(round(float(done) / float(total), 4) * 100)
         ratio = '(' + byte_to_mb(done) + '/' + byte_to_mb(total) + ')'
-        sys.stderr.write(percent + ' ' + ratio + '\r')
+        line = '-> ' + percent + ' ' + ratio
+        sys.stderr.write(line.ljust(c_length) + '\r')
```

Comment:
@MarshallOfSound should this be `last_length` instead of `c_length`?  CI is failing with:
```
Traceback (most recent call last):
  File "script/update", line 367, in <module>
    sys.exit(main())
  File "script/update", line 79, in main
    get_packaged_src(args.clean, git_cache, nohooks, args.upload_packaged_src)
  File "script/update", line 331, in get_packaged_src
    filesystem.download_and_extract(SOURCE_ROOT, package_url)
  File "/home/builduser/project/script/lib/filesystem.py", line 63, in download_and_extract
    sys.stderr.write(line.ljust(c_length) + '\r')
NameError: global name 'c_length' is not defined
```